app/assets/javascripts/host_edit.js

```diff
@@ -221,23 +221,45 @@ function hostgroup_changed(element) {
   var host_id = $("form").data('id');
   var host_changed = $("form").data('type-changed');
   if (host_id) {
-    if (host_changed ){
-      update_form(element,{data:"&host[id]="+host_id});
-    } else if (host_changed == undefined) { // hostgroup changes parent
-      update_form(element);
-    } else { // edit host
-      set_inherited_value(element);
-      update_puppetclasses(element);
-      reload_host_params();
-    }
+    handleHostgroupChangeEdit(element, host_id, host_changed)
   } else { // a new host
-    reset_explicit_values(element);
+    disableComputeResourceDropdown(true);
+    handleHostgroupChangedNew(element).then(function () {
+      disableComputeResourceDropdown(false);
+    });
+  }
+}
+
+function promise() {
```

Comment:
I did not find anything that deals with promises, could you point me to the corresponding place?